File: benchmarks/typescript/src/QueryCounter.ts
/**
 * Counts SQL queries executed during a single benchmark iteration.
 *
 * Usage:
 *   QueryCounter.reset();    // before each iteration
 *   // ... run scenario ...
 *   QueryCounter.get();      // after each iteration
 */
export class QueryCounter {
  private static count: number = 0;

  static reset(): void {
    QueryCounter.count = 0;
  }

  static increment(): void {
    QueryCounter.count++;
  }

  static get(): number {
    return QueryCounter.count;
  }
}

File: benchmarks/java/src/main/java/com/benchmark/QueryCounter.java
package com.benchmark;
 
import java.util.concurrent.atomic.AtomicLong;
 
/**
 * Counts SQL queries executed during a single benchmark iteration.
 *
 * Uses AtomicLong to be safe across Hibernate's internal threading.
 *
 * Usage:
 *   QueryCounter.reset();       // before each iteration
 *   // ... run scenario ...
 *   QueryCounter.get();         // after each iteration
 */
public class QueryCounter {
 
    private static final AtomicLong count = new AtomicLong(0);
 
    public static void reset() {
        count.set(0);
    }
 
    public static void increment() {
        count.incrementAndGet();
    }
 
    public static long get() {
        return count.get();
    }
}
